src/bundle/ui-dev/src/modules/universal-discovery/components/content-create-widget/content.create.widget.js

```diff
@@ -154,44 +191,22 @@ const ContentCreateWidget = () => {
                         </div>
                         <div className="ibexa-instant-filter__desc">{filtersDescLabel}</div>
                         <div className="ibexa-instant-filter__items">
-                            {contentTypesWithSuggestions.map(([groupName, groupItems], index) => {
+                            {contentTypesDataToShow.map(([groupName, groupItems], index) => {
                                 const isSuggestionGroup = !!suggestions.length && index === 0;
-                                const restrictedContentTypeIds = selectedLocation?.permissions?.create.restrictedContentTypeIds ?? [];
-                                const isHiddenGroup = groupItems.every((groupItem) => {
-                                    const isNotSearchedName = filterQuery && !groupItem.name.toLowerCase().includes(filterQuery);
-                                    const hasNotPermission =
-                                        restrictedContentTypeIds.length && !restrictedContentTypeIds.includes(groupItem.id.toString());
-                                    const isNotAllowedContentType =
-                                        allowedContentTypes && !allowedContentTypes.includes(groupItem.identifier);
-                                    const isHiddenByConfig = groupItem.isHidden;
-
-                                    return (
-                                        isNotSearchedName ||
-                                        hasNotPermission ||
-                                        isNotAllowedContentType ||
-                                        isHiddenByConfig ||
-                                        (isNotSearchedName && isSuggestionGroup)
```

Comment:
Note: from what I understand `(isNotSearchedName && isSuggestionGroup)` condition is not needed, because if `isNotSearchedName` is _true_, we will have _return_ and if it is _false_ this condition `(isNotSearchedName && isSuggestionGroup)` is also _false_, so removing `(isNotSearchedName && isSuggestionGroup)` does not affect this condition result.